src/lobby.rs

```diff
@@ -58,13 +66,14 @@ impl Lobby {
             games: inner
                 .games
                 .iter()
-                .map(|(game_id, players)| {
+                .map(|(game_id, lobby)| {
                     (
                         *game_id,
-                        players
+                        lobby
+                            .participants
+                            .clone()
                             .into_iter()
-                            .map(|participant| &participant.player)
-                            .cloned()
+                            .map(|participant| participant.player.clone())
```

Comment:
is there a better way to go from gameLobby -> vec<players>? seems odd that I have to clone twice here, surely i only need a reference to lobby.participants in order to get each member

Reply:
lobby.participants.iter().map(|participant| &participant.player).cloned().collect()

Reply:
`into_iter()` consumes participants and produces an iterator of type `Iterator<Item=Participant>` (which is not allowed because lobby still owns participants). `iter()` does not and produces an iterator of type `Iterator<Item=&Participant>`